spec/helpers/vita_partner_helper_spec.rb

```diff
@@ -2,23 +2,115 @@
 
 describe VitaPartnerHelper do
   describe "#grouped_organization_options" do
-    let(:parent_org1) { create(:vita_partner, name: "First Parent Org") }
-    let(:parent_org2) { create(:vita_partner, name: "Second Parent Org") }
-    let(:parent_org3) { create(:vita_partner, name: "No Child Org") }
-    let(:sub_org1) { create(:vita_partner, parent_organization_id: parent_org1.id, name: "The First Child Org") }
-    let(:sub_org2) { create(:vita_partner, parent_organization_id: parent_org1.id, name: "The Second Child Org") }
-    let(:sub_org3) { create(:vita_partner, parent_organization_id: parent_org2.id, name: "The Third Child Org") }
-
-    it "returns array grouped by parent org" do
-      @vita_partners = VitaPartner.all
-      expected =
-        [
-          ["First Parent Org", [["First Parent Org", parent_org1.id], ["The First Child Org", sub_org1.id], ["The Second Child Org", sub_org2.id]]],
-          ["No Child Org", [["No Child Org", parent_org3.id]]],
-          ["Second Parent Org", [["Second Parent Org", parent_org2.id], ["The Third Child Org", sub_org3.id]]],
+    context "when user's role is Team Member" do
+      let(:team_member) { create :team_member_user }
+
+      before do
+        allow(view).to receive(:current_user).and_return(team_member)
+      end
+
+      it "returns just the user's site and the org" do
+        @vita_partners = VitaPartner.accessible_by(Ability.new(team_member))
+        expected = [
+          [
+            team_member.role.site.parent_organization.name,
+            [[team_member.role.site.name, team_member.role.site.id]]
+          ]
+        ]
+
+        expect(helper.grouped_organization_options).to eq(expected)
+      end
+    end
+
+    context "when the user's role is Site Coordinator" do
+      let(:site_coordinator) { create :site_coordinator_user }
+
+      before do
+        allow(view).to receive(:current_user).and_return(site_coordinator)
+      end
+
+      it "returns just the user's site and the org" do
+        @vita_partners = VitaPartner.accessible_by(Ability.new(site_coordinator))
+        expected = [
+          [
+            site_coordinator.role.site.parent_organization.name,
+            [[site_coordinator.role.site.name, site_coordinator.role.site.id]]
+          ]
         ]
-      expect(helper.grouped_organization_options).to eq(expected)
+
+        expect(helper.grouped_organization_options).to eq(expected)
+      end
     end
 
+    context "when the user's role is Admin" do
```

Comment:
You could group all the tests that have more than one site into one context. Not a blocker.

Consider stubbing `.accessible_vita_partners` in the different situations, rather than doing this by role. That would let you cover the same situations with fewer test cases.